src/firebase.ts

```diff
@@ -65,23 +66,33 @@ export const getStorageApp = (host: string, port: string) => {
   return app;
 };
 
-export function useEmulatedFirebaseApp(name: string, config: any) {
+export function useEmulatedFirebaseApp(
+  name: string,
+  config: any,
+  initialize?: (app: firebase.app.App) => void
+) {
   const projectId = useProjectId();
-
-  const app = useMemo(() => {
```

Comment:
Hooks are called during render phase and should not have side effects like this. Wrapping side effects in `useMemo` won't help. The right way is to `useEffect` for side effects and pair them up with cleanups.

To see the total active apps, run `firestore.app.firebase_.apps` in console. You'll see twice as many apps being created than deleted without the fix.